akka-actor/src/main/scala/akka/actor/dungeon/FaultHandling.scala

```diff
@@ -12,14 +12,27 @@ import akka.dispatch.sysmsg._
 import akka.event.Logging
 import akka.event.Logging.Debug
 import akka.event.Logging.Error
-
 import scala.collection.immutable
 import scala.concurrent.duration.Duration
 import scala.util.control.Exception._
 import scala.util.control.NonFatal
+
 import akka.actor.ActorRefScope
+import akka.annotation.InternalApi
+import com.github.ghik.silencer.silent
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object FaultHandling {
+  sealed trait FailedInfo
+  private case object NoFailedInfo extends FailedInfo
+  private final case class FailedRef(ref: ActorRef, fatally: Boolean) extends FailedInfo
```

Comment:
Would be nice if we could make it simpler, but maybe. I guess you checked that we need both the `FailedRef(, fatally = true)` and `FailedFatally` state.

Reply:
yes, I wanted to change as little as possible, and it seems like we use both the ref and the flag (with more refactoring it might not be needed, but probably not worth changing too much)

Reply:
Superficially, nothing seems to break when just removing the `fatally` flag

Reply:
`fatally` is now gone. I didn't seem to be needed.